sdk/storage/Azure.Storage.Blobs/src/BlobServiceVersionPolicy.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Azure.Core;
+
+namespace Azure.Storage.Blobs
+{
+    internal class BlobServiceVersionPolicy : ServiceVersionPolicy
+    {
+        /// <summary>
+        /// OnSendingRequest override.
+        /// </summary>
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            // Get ServiceVersion header
+            if (!message.Request.Headers.TryGetValue(Constants.HeaderNames.ServiceVersion, out string serviceVersionString))
+            {
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly
+                throw new ArgumentNullException("x-ms-version");
+#pragma warning restore CA2208 // Instantiate argument exceptions correctly
+            }
+
+            // Convert to ServiceVersion
+            BlobClientOptions.ServiceVersion serviceVersion = ToServiceVersion(serviceVersionString);
+
+            // Latest ServiceVersion
+            if (serviceVersion == BlobClientOptions.LatestVersion)
+            {
+                return;
+            }
+
+            // 2019-07-07 check
+            if (serviceVersion < BlobClientOptions.ServiceVersion.V2019_07_07)
+            {
+                // Encryption Scope
+                ThrowIfContainsHeader(message, "x-ms-default-encryption-scope", Constants.Blob.Container.CreateOperationName, serviceVersionString);
+                ThrowIfContainsHeader(message, "x-ms-deny-encryption-scope-override", Constants.Blob.Container.CreateOperationName, serviceVersionString);
+                ThrowIfContainsHeader(message, "x-ms-encryption-scope", "any API", serviceVersionString);
+
+                // Previous Snapshot URL
+                ThrowIfContainsHeader(message, "x-ms-previous-snapshot-url", Constants.Blob.Page.GetManagedDiskPageRangesDiffOperationName, serviceVersionString);
+            }
+        }
+
+        private static BlobClientOptions.ServiceVersion ToServiceVersion(string serviceVersionString)
```

Comment:
This should be shared across services in StorageVersionExtensions.cs and throw if passed an unknown value.

Reply:
BlobClientOptions.ServiceVersion and ShareClientOptions.ServiceVersion are different enums.

Reply:
Yes, but the [aliasing](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/Shared/StorageVersionExtensions.cs#L8) takes care of that.  Have a look at the [shared ToString](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/Shared/StorageVersionExtensions.cs#L47).